### DISCUSSION TITLE
The game capacity is not enough

### DISCUSSION THREAD

**Discussion by semoki** [airsdk/Adobe-Runtime-Support#3437] [Q&A]
The game capacity is not enough. I am reluctant to use graphics resources due to Google's 150 MB limit. I know there is a PAD option, but it is too difficult for me to use.

I place PNG files on the timeline as movie clips and use them for animation.

Which is more advantageous: placing the PNG resources on the server and having them downloaded at first run, or putting the SWF on the server and downloading it?

I am paying for the server and using the server. I can simply use PAD using the server.
Can you tell me how to use the effect or a link?

**Comment by ajwfrost** [comment]
I'm not sure Play Asset delivery is that complicated .. in case it helps, there's a sample for this at https://github.com/airsdk/Adobe-Runtime-Support/tree/master/samples/android_play_asset_delivery
If using Play Asset delivery then it's all done via the play store so you wouldn't need any personal server. You package everything up into a single App Bundle file which can be massive, and then upload that to Google, and they then split it apart and delivery the right things to the right phones at the right time. It does mean you can have different asset packs for different resolutions etc...

But equally, yes if you have your own server you could just download the assets at run-time. It depends then whether you want them to be publicly available and concerns about people taking them to re-use.

I would suspect your usage would need an initial PNG or two to use, so it might be best to have the rest of the PNGs on the server and download them in the background into your app-storage location so you can then load them from there later on. Someone might already have a kind of asset manager thing for this scenario open-sourced..?

**Reply by semoki** [reply]
Thanks for your reply. I will study hard based on the answers you sent. I think it would be best to use a PAD. I would like to ask you one question.
I'm currently using Animate to create AAB, and I'm only using Animate. If you only use animate, there are no additional restrictions or impossibilities, right?

The concern is that PAD cannot be used when uploading to Apple or creating a PC version in the future, so in terms of versatility, it may be better to download additional resources using a personal server.

I'm worried. Anyway, your sincere answers are always the best. Stay healthy always.

**Reply by ajwfrost** [reply]
>  If you only use animate, there are no additional restrictions or impossibilities, right?

Well ... there are definitely extra hoops to jump through if you use Animate, because they have no built-in support for Play Asset Delivery. So you would need to manually configure your application descriptor XML file and you would need to update a configuration file to tell AIR Developer Tool to generate an Android App Bundle (when Animate tells it to create an APK file)..

> The concern is that PAD cannot be used when uploading to Apple or creating a PC version in the future, so in terms of versatility, it may be better to download additional resources using a personal server.

That is a very good point! There would definitely need to be significantly different code for the Android-specific asset retrieval. And if you're going to need the resource server for other platforms anyway, then there seems no point to have Android work differently. One point to note: if using iOS, then you wouldn't be able to load in normal SWF files at run-time, so best to keep all the graphics as separate PNG/etc files..

thanks
